[jit] add top-level readme to csrc/jit

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #33919 [jit] fix up refs in overview.md
* **#33916 [jit] add top-level readme to csrc/jit**

Differential Revision: [D20150771](https://our.internmc.facebook.com/intern/diff/D20150771)

suo committed:
[jit] add top-level readme to csrc/jit

[ghstack-poisoned]

driazati commented on torch/csrc/jit/README.md:
```diff
@@ -0,0 +1,25 @@
+- `python/`: Binding stuff into Python or accessing information from the Python environment.
+- `testing/`: Utilities and helpers for testing.
+- `mobile/`: Mobile-specific implementations of runtime components.
+- `passes/`: IR-to-IR passes, generally for optimization and lowering.
```
Can you add a bit about the existing `backends`, `generated`, `codegen` `tensorexpr`, and `testing` folders

suo replied:
I think `codegen` and `testing` are already documented. I plan to move `tensorexpr` under `codegen`. And `backends` is to be deleted shortly. I'll add a note about `generated`.

suo committed:
Update on "[jit] add top-level readme to csrc/jit"


Differential Revision: [D20150771](https://our.internmc.facebook.com/intern/diff/D20150771)

[ghstack-poisoned]
